### PR TITLE
reordered apireference.md to clarify basis statuses

### PR DESCRIPTION
**Pull request by chriscoey**

### CODE REVIEW

File: src/MathOptInterface.jl [hunk 1]
```diff
@@ -31,7 +31,7 @@ Start the solution procedure.
 function optimize! end
 
 """
-    freesolver instance!(m::AbstractSolverInstance)
+    freesolverinstance!(m::AbstractSolverInstance)
```
Comment (mlubin) [inline src/MathOptInterface.jl]: I'd vote to rename this to ``free!``

Comment (chriscoey) [inline src/MathOptInterface.jl]: i was thinking the same

Comment (chriscoey) [inline src/MathOptInterface.jl]: changing it

File: docs/src/apireference.md [hunk 2]
```diff
@@ -96,47 +96,18 @@ The value of the attribute is of type `ResultStatusCode`.
 ResultStatusCode
 ```
 
+## Variables and Constraints
+
 ### Basis Status
 
-[what? BasisStatus]
+The `BasisStatus` attribute of a variable or constraint indicates its presence in a basis.
```
Comment (mlubin) [inline docs/src/apireference.md]: There's more than one way in which a variable/constraint may be *not* in the basis (i.e., at an upper or lower bound), so this wording is a bit unclear.

Comment (chriscoey) [inline docs/src/apireference.md]: reasonable

Comment (chriscoey) [inline docs/src/apireference.md]: fixed but i still think more thought needs to go into these. the basis status description in attributes.jl is not super clear. eg what if it's not an optimal primal solution, but rather a ray

**Pull request merged**